projects/templates/src/lib/components/po-page-dynamic-detail/po-page-dynamic-detail.component.spec.ts

```diff
@@ -158,121 +166,288 @@ describe('PoPageDynamicDetailComponent:', () => {
       component.model = { id: 1 , name: 'Angular' };
 
       spyOn(component, <any> 'formatUniqueKey').and.returnValue(uniqueKey);
-      spyOn(component['poPageDynamicService'], 'deleteResource').and.returnValue(getObservable({}));
+      spyOn(component['poPageDynamicService'], 'deleteResource').and.returnValue(of({}));
       spyOn(component, <any> 'navigateTo');
       spyOn(component['poNotification'], 'success');
 
       component['remove'](path);
 
-      tick(50);
+      tick();
 
       expect(component['poNotification'].success).toHaveBeenCalled();
       expect(component['formatUniqueKey']).toHaveBeenCalledWith(component.model);
       expect(component['poPageDynamicService'].deleteResource).toHaveBeenCalledWith(uniqueKey);
       expect(component['navigateTo']).toHaveBeenCalledWith({ path/*, component: PoPageDynamicTableComponent*/ });
     }));
 
-    it('ngOnInit: should call `loadData` with `paramId` if `activatedRoute.snapshot.data.serviceApi` is falsy', () => {
-      const id = 1;
-      const activatedRoute: any = {
-        snapshot: {
-          data: { },
-          params: { id }
-        }
-      };
+    describe('ngOnInit:', () => {
+      it('should call `loadData` with `paramId` if `activatedRoute.snapshot.data.serviceApi` is falsy', () => {
+        const id = 1;
+        const activatedRoute: any = {
+          snapshot: {
+            data: { },
+            params: { id }
+          }
+        };
 
-      component.serviceApi = 'localhost:4300/api/people';
+        component.serviceApi = 'localhost:4300/api/people';
 
-      spyOn(component, <any> 'loadData');
-      spyOn(component['poPageDynamicService'], <any> 'configServiceApi');
+        spyOn(component, <any> 'getMetadata').and.returnValue(of({}));
+        spyOn(component, <any> 'loadData').and.returnValue(of({}));
+        spyOn(component, <any> 'loadOptionsOnInitialize').and.returnValue(of({}));
+        spyOn(component['poPageDynamicService'], <any> 'configServiceApi');
+
+        component['activatedRoute'] = activatedRoute;
 
-      component['activatedRoute'] = activatedRoute;
+        component.ngOnInit();
 
-      component.ngOnInit();
+        expect(component['loadData']).toHaveBeenCalledWith(id);
+        expect(component['poPageDynamicService'].configServiceApi)
+          .toHaveBeenCalledWith({ endpoint: component.serviceApi, metadata: undefined });
+      });
 
-      expect(component['loadData']).toHaveBeenCalledWith(id);
-      expect(component['poPageDynamicService'].configServiceApi).toHaveBeenCalledWith({ endpoint: component.serviceApi });
-    });
+      it('should call `loadMetadata` with `id` and set `serviceApi` if `activatedRoute.snapshot.data.serviceApi` is truthy',
```

Comment:
- Este método `loadMetadata` é o `loadData`?
- Existe uma condição que na qual `loadMetadata` só é chamado se o serviceApi for preenchido?